### PR TITLE
feat(atomizer): updating the box alignment logic

### PR DESCRIPTION
**Pull request by carriemorrison**
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

---

### Description

*W3C has been developing the Box Alignment Module, which establishes a common set of alignment properties and values for use across multiple box models, including flex, grid, table and block.*
I like where they are going with grouping alignments so makes sense (to me) to follow their syntax.

e.g.

*Value:* `normal | <baseline-position> | <content-distribution> | <overflow-position>? <content-position>`

# [Box alignment model](https://www.w3.org/TR/css-align-3/)

[ ]  *baseline position*: `first baseline` & `last baseline` not ready for use
https://www.w3.org/TR/2021/WD-css-align-3-20211224/#typedef-baseline-position
https://caniuse.com/mdn-css_properties_align-items_flex_context_last_baseline

[x] *Content Distribution*
https://www.w3.org/TR/2021/WD-css-align-3-20211224/#typedef-content-distribution

[ ] *overflow-position*: not ready for use
https://www.w3.org/TR/2021/WD-css-align-3-20211224/#typedef-overflow-position
https://caniuse.com/mdn-css_properties_align-items_flex_context_safe_unsafe

[ ] *content-position*: `start` and `end` not ready for use
https://www.w3.org/TR/2021/WD-css-align-3-20211224/#typedef-content-position
https://caniuse.com/mdn-css_properties_align-items_flex_context_start_end

[x] *self-position*: content-position + `self-end` & `self-start`
https://www.w3.org/TR/2021/WD-css-align-3-20211224/#typedef-self-position

[ ] Legacy values for justify-items - they may get dropped..

## Added helpers to match the updated specs eg 

```
const { baselinePosition, contentDistribution, contentPosition } = require('./boxAlignment');
const selfPosition = {...contentPosition, 'se': 'self-end', 'ss': 'self-start'};
```
**note:** I commented out some of the unusable values in these helpers - we could simply remove them for now too...

### Output

```
baselinePosition { 
    'b': 'baseline' 
}
contentDistribution {
    'sa': 'space-around',
    'sb': 'space-between',
    'se': 'space-evenly',
    'st': 'stretch'
}
contentPosition { 
    'c': 'center',
    'fe': 'flex-end',
    'fs': 'flex-start'
}
selfPosition {
    'c': 'center',
    'fe': 'flex-end',
    'fs': 'flex-start',
    'se': 'self-end',
    'ss': 'self-start'
}
```

**note: Due to using spread operator to combine rules the AFTER output below is not alphasorted **

## [align-content](https://www.w3.org/TR/css-align-3/#propdef-align-content)

Value:	`normal | <baseline-position> | <content-distribution> | <overflow-position>? <content-position>`

### Before

```
'arguments': [{
    'fs': 'flex-start',
    'fe': 'flex-end',
    'c': 'center',
    'sb': 'space-between',
    'sa': 'space-around',
    'st': 'stretch'
}]
```

### After

```
'arguments': [{
    'b': 'baseline',
    'n': 'normal',
    'sa': 'space-around',
    'sb': 'space-between',
    'se': 'space-evenly',
    'st': 'stretch',
    'c': 'center',
    'fe': 'flex-end',
    'fs': 'flex-start'
}]
```

## [align-items](https://www.w3.org/TR/css-align-3/#propdef-align-items)

[Css-tricks](https://css-tricks.com/almanac/properties/j/justify-items/)

Value: `normal | stretch | <baseline-position> | [ <overflow-position>? <self-position> ]`

### Before

```
'arguments': [{
    'fs': 'flex-start',
    'fe': 'flex-end',
    'c': 'center',
    'b': 'baseline',
    'st': 'stretch'
}]
```

### After

```
'arguments': [{
    'n': 'normal',
    'st': 'stretch',
    'b': 'baseline',
    'c': 'center',
    'fe': 'flex-end',
    'fs': 'flex-start'
}]
```

## [align-self](https://www.w3.org/TR/css-align-3/#propdef-align-self)

Value:	`auto | normal | stretch | <baseline-position> | <overflow-position>? <self-position>`

### Before

```
'arguments': [{
    'a': 'auto',
    'fs': 'flex-start',
    'fe': 'flex-end',
    'c': 'center',
    'b': 'baseline',
    'st': 'stretch'
}]
```

### After

```
'arguments': [{
    'a': 'auto',
    'n': 'normal',
    's': 'stretch',
    'b': 'baseline',
    'c': 'center',
    'fe': 'flex-end',
    'fs': 'flex-start',
    'se': 'self-end',
    'ss': 'self-start'
}]
```

## [justify-content](https://www.w3.org/TR/css-align-3/#propdef-justify-content)

Value: `normal | <content-distribution> | <overflow-position>? [ <content-position> | left | right ]`

### Before

```
'arguments': [{
    /* Positional alignment */
    'fs': 'flex-start',
    'fe': 'flex-end',
    'c': 'center',
    
    /* Distributed alignment */
    'sb': 'space-between',
    'sa': 'space-around',
    'se': 'space-evenly',
    's':  'stretch'
}]
```

### After

```
'arguments': [{
    'n': 'normal',
    'l': 'left',
    'r': 'right',
    'sa': 'space-around',
    'sb': 'space-between',
    'se': 'space-evenly',
    'st': 'stretch',
    'c': 'center',
    'fe': 'flex-end',
    'fs': 'flex-start'
}]
```

# [Isolation](https://www.w3.org/TR/compositing-1/#propdef-isolation)

Pretty simple this one.

## TODO

Add `Justify-items` for flex layout.

Move Align properties to be grouped, kept current order for easier initial PR..

### How Has This Been Tested?

_Please describe in detail how you tested your changes. Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code, etc._

### Types of changes

What types of changes does your code introduce? Put an x in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

Review the following points, and put an x in all the boxes that apply. If you’re unsure about any of these, don’t hesitate to ask. We’re here to help!

- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/acss-io/atomizer/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by carriemorrison** [comment]
Is there any benefit in updating/adding the `example/html/sample-*.html` files?

**Comment by carriemorrison** [comment]
Screen Shot 2022-07-19 at 11 00 18 AM

**Comment by carriemorrison** [comment]
localhost_3000_align html

File: packages/atomizer/src/rules.js [hunk 1]
```diff
@@ -1777,6 +1772,24 @@ module.exports = [
...
+    },
+        /**
+    ==================================================================
+    ISOLATE
+    ==================================================================
+    */
+    {
+        'type': 'pattern',
+        'name': 'Isolation',
+        'matcher': 'I',
```
Comment (kfay) [inline packages/atomizer/src/rules.js:1784]: `I` is a pretty generic abbreviation for this property. What do you think about something like `Iso` instead?

Comment (carriemorrison) [reply packages/atomizer/src/rules.js:1784]: Sounds good to me :)

File: examples/html/sample-2.html [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<div class="Bgbm(n) Mbm(pl)">Test Blend modes</div>
```
Comment (redonkulus) [inline examples/html/sample-2.html:1]: This file should be removed

**Comment by carriemorrison** [comment]
Comparing the new rules to previous: 

Screen Shot 2022-07-20 at 1 41 24 PM

there is 1 mismatch with `s`: `stretch` which is a little annoying.

We could dupe, or I just stick with simply adding the rules for `Justify-content` like it was before.

```
    {
        'type': 'pattern',
        'name': 'Justify content',
        'matcher': 'Jc',
        'allowParamToValue': false,
        'styles': {
            'justify-content': '$0'
        },
        'arguments': [{
            'n': 'normal',
            'l': 'left',
            'r': 'right',
            's': 'stretch', // backwards compat
            ...contentDistribution,
            ...contentPosition
        }]
    },
```

Screen Shot 2022-07-20 at 1 39 40 PM


Also would be nice to be able to alpha sort the reference, will look into it..

**Comment by carriemorrison** [comment]
**Comparing the reference lookups, left is prod, right is localhost:**

Screen Shot 2022-07-20 at 2 01 01 PM

Screen Shot 2022-07-20 at 2 01 30 PM

Screen Shot 2022-07-20 at 2 01 44 PM

Screen Shot 2022-07-20 at 2 02 17 PM

**Review by redonkulus — APPROVED** [review]
Changes look good to me. Thanks for updating the docs.

**Pull request merged**